app/services/scoring/player_svc.rb

```diff
@@ -0,0 +1,19 @@
+class Scoring::PlayerSvc
+
+    #calculates the points earned by player in match
+    def self.call(player, match)
+
+        #list of all of player's player_stats from match
+        player_stats = match.player_stats.select {|stat| stat.player_id == player.id}
```

Comment:
better to do this all with an active record query. as you have it, you'll be evaluating all the player_stats from the match, but you could get them with one query like `PlayerStat.where(match_id: match.id, player_id: player.id)` or `match.player_stats.where(player_id: player.id)` either will do the same thing.